packages/@aws-cdk/aws-scheduler-targets-alpha/lib/sage-maker-start-pipeline-execution.ts

```diff
@@ -0,0 +1,96 @@
+import { ISchedule, IScheduleTarget, ScheduleTargetConfig } from '@aws-cdk/aws-scheduler-alpha';
+import { ArnFormat, Names, Stack } from 'aws-cdk-lib';
+import { IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { CfnPipeline } from 'aws-cdk-lib/aws-sagemaker';
+import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
+import { sameEnvDimension } from './util';
+
+/**
+ * Properties for a pipeline parameter
+ */
+export interface SageMakerPipelineParameter {
+  /**
+   * Name of parameter to start execution of a SageMaker Model Building Pipeline.
+   */
+  readonly name: string;
+
+  /**
+   * Value of parameter to start execution of a SageMaker Model Building Pipeline.
+   */
+  readonly value: string;
+}
+
+/**
+ * Properties for a SageMaker Target
+ */
+export interface SageMakerStartPipelineExecutionProps extends ScheduleTargetBaseProps {
+  /**
+   * List of parameter names and values to use when executing the SageMaker Model Building Pipeline.
+   *
+   * The length must be between 0 and 200.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-scheduler-schedule-sagemakerpipelineparameters.html#cfn-scheduler-schedule-sagemakerpipelineparameters-pipelineparameterlist
+   *
+   * @default - no pipeline parameter list
+   */
+  readonly pipelineParameterList?: SageMakerPipelineParameter[];
+}
+
+/**
+ * Use a SageMaker pipeline as a target for AWS EventBridge Scheduler.
+ */
+export class SageMakerStartPipelineExecution extends ScheduleTargetBase implements IScheduleTarget {
+  private readonly pipelineArn: string;
+
+  constructor(
+    private readonly pipeline: CfnPipeline,
```

Comment:
Everything is fine except for this... it should be an `IPipeline` but that doesn't exist yet. The problem is that this locks in people to the L1 pipeline, and ideally we'd want this to be interoperable with a community L2 if that existed.

So how about adding a very barebones `IPipeline` construct to `aws-sagemaker`. Nothing controversial, just kind of a placeholder. In the future, if we develop a Pipeline L2, it can build off of the `IPipeline` as can any community L2. As long as we don't make any crazy decisions, this can be stable from day 1. We did something similar in `IEndpoint`.

Reply:
Good idea.
The `IEndpoint` is also creating an interface in the `aws-sagemaker-alpha` module, but that is to ensure that the existing implementation is not affected, so is it correct to create it only in the `aws-sagemaker` module for this PR?

Reply:
I tried to change it.

https://github.com/aws/aws-cdk/pull/28927/commits/6ce1b25c114f1197573c56eae210c4e98a3b29c1

Reply:
I changed.

https://github.com/aws/aws-cdk/pull/28927/commits/43a0626e03e319df589b59afd7db47082dac19d9

Reply:
I changed.
https://github.com/aws/aws-cdk/pull/28927/commits/0e29910efa75d549c8f2273757770ef17f3cc13d

Reply:
I changed.